### PR TITLE
feat(logger): add basic instrumentation to the API

### PR DESCRIPTION
**Pull request by oddgrd**
## Description of change

Add some basic instrumentation to the logger API. This should also allow us to observe the volume of logs sent from specific deployments.

## How has this been tested? (if applicable)

Tested locally and with honeycomb test environment.

### CODE REVIEW

**Review by orhun — APPROVED** [review]

File: logger/src/lib.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ where
     }
 
     async fn get_logs(&self, deployment_id: String) -> Result<Vec<LogLine>, Error> {
+        tracing::info!(deployment_id = %deployment_id, "fetching logs for deployment");
```
Comment (chesedo) [inline logger/src/lib.rs:48]: I think we will be fine removing all the info traces in this PR

File: logger/src/lib.rs [hunk 2]
```diff
@@ -56,13 +57,19 @@ impl<D> Logger for Service<D>
...
     async fn store_logs(
         &self,
         request: Request<StoreLogsRequest>,
     ) -> Result<Response<StoreLogsResponse>, Status> {
         let request = request.into_inner();
         let logs = request.logs;
+
         if !logs.is_empty() {
+            Span::current().record("deployment_id", &logs[0].deployment_id);
+            Span::current().record("batch_size", logs.len());
```
Comment (chesedo) [inline logger/src/lib.rs:70]: Maybe store the result of `Span::current()` so that we don't call it multiple times.

**Review by chesedo — COMMENTED** [review]
Some small comments :smile:

File: logger/src/lib.rs (hunk 1 above)
Comment (oddgrd) [reply logger/src/lib.rs:48]: Hmm, will we see anything if there are no events?

Comment (oddgrd) [reply logger/src/lib.rs:48]: Ah, we will see it in Honeycomb, but not in the logger itself. I'll remove the events then, since they are quite verbose.

**Review by chesedo — APPROVED** [review]
LGTM

**Pull request merged**